src/main/java/Parser.java

```diff
@@ -0,0 +1,330 @@
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Reads and processes the commands given to Duke and returns the appropriate action.
+ * Generates either todo, deadline or event tasks.
+ */
+public class Parser {
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Creates a Parser object with the given TaskList, Ui, Storage.
+     * @param taskList TaskList that duke keeps track of.
+     * @param ui Generates the output string.
+     * @param storage Accesses the given file.
+     */
+    public Parser(TaskList taskList, Ui ui, Storage storage) {
+        this.taskList = taskList;
+        this.ui = ui;
+        this.storage = storage;
+    }
+
+    /**
+     * Processes the Command given by the user to Duke.
+     * @param command String representing the command given to Duke.
+     * @return A string representing the reply from Duke.
+     */
+    public String processCommand(String command) {
+        String reply = "";
+
+        // When command is 'list'
+        if (command.equals("list")) {

```

Comment:
Is there a reason why you opted to use multiple if-else blocks? a switch statement might be better choice for such implementations as it implicitly uses a hash table to perform the checks